lib/libutee/arch/arm/ftrace/ftrace.c

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2019, Linaro Limited
+ */
+
+#include <setjmp.h>
+#include <user_ta_header.h>
+#include <utee_syscalls.h>
+#include "ftrace.h"
+
+static const char hex_str[] = "0123456789abcdef";
+static uint32_t ret_idx_setjmp;
+
+static void __noprof ftrace_memmove(struct ftrace_buf *fbuf, size_t size)
```

Comment:
I'd prefer if the name of this function reflected what's it doing. Does it save or make room for something?
Using a `_memmove` suffix is not good since it doesn't handle overlapping memory areas (see `memmove()` vs `memcpy()`).

Reply:
It is similar to what `memmove` does given that `dst < src`. We can't use standard library apis (eg. `memmove()`) here as those can be profiled too leading to a circular loop, so I have to create `ftrace_memmove` api here as `__noprof`.

Reply:
It takes different arguments than `memmove()` so it can't do the same thing. What does this function do? What its purpose?

Reply:
It basically shifts/moves ftrace buffer to create space for new dump in case the buffer size falls short. Would you prefer `fbuf_shift()`?

Reply:
Yes, perhaps with some comment too.